solutions/advanced/cses-1655.mdx

```diff
@@ -0,0 +1,89 @@
+---
+id: cses-1655
+source: CSES
+title: Maximum Xor Subarray
+author: Mihnea Brebenel
+---
+
+## Explanation
+
+Let $\texttt{pref}_i$ be the xor-sum of the prefix ending at $i$. Then, we must find a previous prefix sum
+$\texttt{pref}_j$ such that $\texttt{pref}_i \oplus \texttt{pref}_j$ is maximized. To obtain the maximal result
+the $0$ bits should be flipped to $1$. And this is where a trie data structure comes in handy. It can be used
+to maintain the binary representations of the xor-sums of prefixes.
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(N)$
```

Comment:
technically the max depth of the trie is constant, but I think it would be good to include a log factor b/c of the depth

Reply:
oki doki